src/screens/IndexScreen.js

```diff
@@ -159,6 +165,28 @@ export const IndexScreen = ({ navigation, route }) => {
     setQueryVariables(variables);
   }, [excludeDataProviderIds, query, route.params?.queryVariables]);
 
+  useEffect(() => {
+    // if we filter for daily events we need to update the query variables with a date range of today
+    if (filterByDailyEvents) {
+      setQueryVariables((prevQueryVariables) => {
+        // remove a refetch key if present, which was necessary for unselecting daily events
+        delete prevQueryVariables.refetch;
+
+        // add the filter key for the specific query, when filtering for daily events
+        return { ...prevQueryVariables, dateRange: [currentDate, currentDate] };
+      });
+    } else {
+      setQueryVariables((prevQueryVariables) => {
+        // remove the filter key for the specific query, when unselecting daily events
+        delete queryVariables.dateRange;
+        // need to spread the `prevQueryVariables` into a new object with additional refetch key
+        // to force the Query component to update the data, otherwise it is not fired somehow
+        // because the state variable wouldn't change
+        return { ...prevQueryVariables, refetch: true };
+      });
+    }
+  }, [filterByDailyEvents]);
+
```

Comment:
when I switch the filter off and on, the events load late. This may be due to internet connection speed, I'm not sure. But it doesn't look good when the list is refreshed late.

https://user-images.githubusercontent.com/11755668/180493941-d70c5f86-5713-4842-a384-5e635aa70c7d.mov

Reply:
interesting, you mean the behavior at the end, right, where there is no loading spinner when toggling off. did this happen on different devices? at my tests while development the loading spinner took place in both cases, when switching on and also off.

Reply:
Was there any activity event while you were doing this test? Because I got this error after setting a new event for a specific day.

Reply:
> after setting a new event for a specific day

what does this mean? i did nothing special, just opened the app and toggled the switch with the outcomes of my screenshots and the white screen with loading indicator in between.

Reply:
Today `25.07.2022`. When we activate this filter, it only shows today's events. If there is no event today, it works fine. But if there is an event today, this problem occurs.

Reply:
Ok thanks, I understand and will check more on this.

Reply:
please check again now. there are several changes according behavior and rendering. there is no more "white screen with loading" when filtering but a loading spinner inside the view under the filters.
with refactoring stuff i hope the problem of weird list item rendering on toggling for daily events is gone.